website/public_adventures.py

```diff
@@ -245,18 +249,13 @@ def clone_adventure(self, user, adventure_id):
                     break
         return render_partial('htmx-adventure-card.html', user=user, adventure=adventure, level=level,)
 
-    def update_filters(self, adventures,  to_filter):
-        if to_filter == 'lang':
-            self.available_languages = set()
-        else:
-            self.available_tags = set()
-        for adventure in adventures:
-            if to_filter == 'lang':
-                adv_lang = adventure.get("language", g.lang)
-                self.available_languages.update([adv_lang])
-            else:
-                adv_tags = adventure.get("tags", [])
-                self.available_tags.update(adv_tags)
+    def update_lang_filter(self, adventures):
+        langs = [a['language'] for a in adventures if a.get('language')]
+        self.available_languages = set(langs)
```

Comment:
It scares me a little that these seem to be instance members that go into a Flask template.

At best, they get overwritten on every request, and them being instance members is not necessary.

At worst, we write code expecting them to persist between requests and (because it works on a local dev setup) we ship that code and now the code behaves basically randomly on the production server.

Are we sure that these won't pollute into different requests?